### PR TITLE
fix(ci): Install hadolint in pre-commit workflow

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/637

This fixes two things:

- Install the hadolint binary to be able to run the pre-commit hook
- Set the default language version of `node` to `system`. This fixes issues on NixOS systems.

### CODE REVIEW

File: .github/workflows/pr_pre-commit.yml [hunk 1]
```diff
@@ -18,10 +19,22 @@ jobs:
       - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
         with:
           python-version: '3.12'
-      - uses: dtolnay/rust-toolchain@master
```
Comment (NickLarsenNZ) [inline .github/workflows/pr_pre-commit.yml]: Is this not required? I thought it was, for clippy, etc...

Comment (Techassi) [reply .github/workflows/pr_pre-commit.yml]: Yeah, it is required. I accidentally removed it.

Comment (Techassi) [reply .github/workflows/pr_pre-commit.yml]: Re-added in 75fed3e

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**